### PR TITLE
@/ui Switch to dependencies catalogs

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

To help us better deal with breaking changes on dependency upgrades, this PR migrates our workspace to use PNPM [catalogs](https://pnpm.io/catalogs). Most apps will use the default catalog (with the most up-to-date dependencies) but some apps will use the mui-styles catalog as they won't be able to make the switch to MUI 6 as is.

NOTE: There appears to be a [bug/issue](https://github.com/babel/babel/pull/16781) in Babel that breaks our jest testing. This PR will be in draft mode until that issue is resolved (or someone adventurous enough figures out which exact dependency is still using the old `syntax-import-assertions` plugin.

Update: PR above has been merged... just waiting for [it to be released](https://github.com/babel/babel/releases/) now.

Update: PR was released as part of new babel release last week but unfortunately, Next.js via `next/babel` preset bundles old versions of babel and hence until they release a new Next.js version, we'll move ahead with a temporary fix suggested [here](https://github.com/babel/babel/issues/16798#issuecomment-2326220768).

## Type of change

- [x] Chore

## Screenshots

N/A

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/jest-config-commons-ui/__mocks__/nextFontMock.js
Comment (DavidTheProgrammer) [inline packages/jest-config-commons-ui/__mocks__/nextFontMock.js:1]: Are these files supposed to be in this PR?

**Review by DavidTheProgrammer — COMMENTED** [review]
Reading the docs, this looks good and dedupes deps. 👍🏾

File: packages/jest-config-commons-ui/__mocks__/nextFontMock.js
Comment (kilemensi) [reply packages/jest-config-commons-ui/__mocks__/nextFontMock.js:1]: Yebo [yes](https://nextjs.org/docs/pages/building-your-application/testing/jest#handling-fonts)

Comment (DavidTheProgrammer) [reply packages/jest-config-commons-ui/__mocks__/nextFontMock.js:1]: Looks good. Just need to fix the build and we're good to go 🚀

**Review by DavidTheProgrammer — APPROVED** [review]
LGTM 🚀

**Review by kelvinkipruto — APPROVED** [review]

**Review by m453h — APPROVED** [review]

**Review by VinneyJ — APPROVED** [review]

**Pull request merged**